src/caffe/net.cpp

```diff
@@ -40,6 +40,22 @@ Net<Dtype>::Net(const string& param_file, Phase phase,
   Init(param);
 }
 
+template <typename Dtype>
+Net<Dtype>::Net(std::istream& param_stream, const string& name_to_log, Phase phase,
+  const int level, const vector<string>* stages) {
+  NetParameter param;
+  ReadNetParamsFromTextStreamOrDie(param_stream, name_to_log, &param);
+  // Set phase, stages and level
+  param.mutable_state()->set_phase(phase);
+  if (stages != NULL) {
```

Comment:
Why not if(stages)? Why not nullptr? NULL is obsolete.

Reply:
fixed